admin/class-yoast-notification.php

```diff
@@ -413,4 +422,22 @@ private function normalize_options( $options ) {
 	private function parse_attributes( &$value, $key ) {
 		$value = sprintf( '%s="%s"', sanitize_key( $key ), esc_attr( $value ) );
 	}
+
+	/**
+	 * Unsets user field if present and set user_id.
+	 *
+	 * @internal This is meant to be used by the Yoast SEO plugin only.
+	 *
+	 * @return bool If the user field was present.
+	 */
+	public function user_to_user_id() {
+		if ( array_key_exists( 'user', $this->options ) ) {
+			// No check needed as we call this once the notification has already been stored.
+			$user_id = $this->options['user']->ID;
+			unset( $this->options['user'] );
+			$this->options['user_id'] = $user_id;
```

Comment:
I don't think there is a need for the intermediate variable? Though are we sure the user is always a WP_User and has ID?
```suggestion
			$this->options['user_id'] = $this->options['user']->ID;
			unset( $this->options['user'] );
```

Reply:
> Though are we sure the user is always a WP_User and has ID?
I'd say so: we call the `user_to_user_id` method in the `update_storage` method of the `Notification_Center`, which works on a `Notification[]`. in the `Notification` constructor we previously made sure that .1 `$options['user']` exists by enforcing the options defaults, and 2. it is not null.